nano/test_common/ledger.cpp

```diff
@@ -1,12 +1,19 @@
 #include <nano/node/node.hpp>
 #include <nano/test_common/ledger.hpp>
 
-nano::test::context::ledger_context::ledger_context () :
+nano::test::context::ledger_context::ledger_context (std::deque<std::shared_ptr<nano::block>> && blocks) :
 	store_m{ nano::make_store (logger, nano::unique_path (), nano::dev::constants) },
-	ledger_m{ *store_m, stats_m, nano::dev::constants }
+	ledger_m{ *store_m, stats_m, nano::dev::constants },
+	blocks_m{ blocks }
 {
 	debug_assert (!store_m->init_error ());
-	store_m->initialize (store_m->tx_begin_write (), ledger_m.cache, ledger_m.constants);
+	auto tx = store_m->tx_begin_write ();
+	store_m->initialize (tx, ledger_m.cache, ledger_m.constants);
+	for (auto const & i : blocks_m)
+	{
+		auto process_result = ledger_m.process (tx, *i);
+		debug_assert (process_result.code == nano::process_result::progress);
+	}
 }
```

Comment:
Should you wait until the blocks are confirmed?

Reply:
I think we have tests that need both so a way to select if the initialisation set should be confirmed seems appropriate future work.